src/controllers/main/main.ts

```diff
@@ -1010,19 +1010,34 @@ export class MainController extends EventEmitter {
     })
   }
 
-  async #ensureAccountInfo(accountAddr: AccountId, networkId: NetworkId) {
+  async #ensureAccountInfo(
+    accountAddr: AccountId,
+    networkId: NetworkId
+  ): Promise<{ hasAccountInfo: true } | { hasAccountInfo: false; errorMessage: string }> {
     await this.#initialLoadPromise
     // Initial sanity check: does this account even exist?
     if (!this.accounts.accounts.find((x) => x.addr === accountAddr)) {
-      this.signAccOpInitError = `Account ${accountAddr} does not exist`
-      return
+      return {
+        hasAccountInfo: false,
+        errorMessage: `Account ${accountAddr} does not exist`
+      }
     }
     // If this still didn't work, re-load
     if (!this.accounts.accountStates[accountAddr]?.[networkId])
       await this.accounts.updateAccountState(accountAddr, 'pending', [networkId])
     // If this still didn't work, throw error: this prob means that we're calling for a non-existent acc/network
-    if (!this.accounts.accountStates[accountAddr]?.[networkId])
-      this.signAccOpInitError = `Failed to retrieve account info for ${networkId}, because of one of the following reasons: 1) network doesn't exist, 2) RPC is down for this network`
+    if (!this.accounts.accountStates[accountAddr]?.[networkId]) {
+      const networkName = networkId[0].toUpperCase() + networkId.slice(1)
```

Comment:
Get the name from this.networks.networks and fallback to the networkId
![image](https://github.com/user-attachments/assets/910db0e8-1688-4bac-a044-665674405a05)

Reply:
Good catch! I didn' want to overcomplicate it, but didn't consider that we have network ids having a dash.